src/main/kotlin/com/projectsurvival/serializing/ConfigLoader.kt

```diff
@@ -0,0 +1,112 @@
+package com.projectsurvival.serializing
+
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import com.google.gson.JsonElement
+import com.mojang.serialization.DynamicOps
+import com.mojang.serialization.JsonOps
+import com.projectsurvival.configs.TestConfig
+import net.fabricmc.loader.api.FabricLoader
+import net.minecraft.nbt.NbtIo
+import net.minecraft.nbt.NbtOps
+import net.minecraft.registry.DynamicRegistryManager
+import net.minecraft.registry.RegistryOps
+import org.kodein.di.DI
+import org.kodein.di.bindSingleton
+import java.io.DataOutputStream
+import java.io.File
+import kotlin.io.path.*
+import kotlin.reflect.full.companionObjectInstance
+import kotlin.reflect.full.primaryConstructor
+
+class ConfigLoader(
+    val configDirectory: File,
+    val registryAccess: DynamicRegistryManager.Immutable
+) {
+    val gson: Gson = GsonBuilder().setPrettyPrinting().create()
+
+    inline fun <reified C : CodecSerializable<C>> loadConfig(path: String, rwIO: ConfigRWComparator<C>): C {
+        val file = File(FabricLoader.getInstance().configDir.absolutePathString(), path)
+
+        println(file.absolutePath)
```

Comment:
I'm not sure is a good idea to print anything to `stdout`. I suggest write this line using logger.